parsl/monitoring/monitoring.py

```diff
@@ -433,10 +448,17 @@ def start(self, priority_msgs, node_msgs, resource_msgs):
         self.logger.info("Hub finished")
 
 
-def hub_starter(comm_q, priority_msgs, node_msgs, resource_msgs, *args, **kwargs):
+def hub_starter(comm_q, exception_q, priority_msgs, node_msgs, resource_msgs, *args, **kwargs):
     hub = Hub(*args, **kwargs)
     comm_q.put((hub.hub_port, hub.ic_port))
-    hub.start(priority_msgs, node_msgs, resource_msgs)
+    hub.logger.info("Starting Hub in Hub starter")
+    try:
+        hub.start(priority_msgs, node_msgs, resource_msgs)
+    except Exception:
+        hub.logger.exception("hub.start exception")
+        exception_q.put("Hub got exception")
```

Comment:
Could we add exception message itself so that the main process can report it better ?

Reply:
Fixed this.